packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js

```diff
@@ -11,14 +16,103 @@ const Separator = styled(Toolbar.Separator)`
   height: ${pxToRem(24)};
 `;
 
+const ModifierButton = ({ icon, name, label }) => {
+  const editor = useSlate();
+
+  /**
+   * @param {string} name - name of the modifier
+   */
+  const isModifierActive = (name) => {
+    const modifiers = Editor.marks(editor);
+
+    return modifiers ? modifiers[name] === true : false;
+  };
+
+  /**
+   * @param {string} name - name of the modifier
+   */
+  const toggleModifier = (name) => {
+    const isActive = isModifierActive(name);
+
+    if (isActive) {
+      Editor.removeMark(editor, name);
+    } else {
+      Editor.addMark(editor, name, true);
+    }
+  };
+  const isActive = isModifierActive(name);
+
+  const { formatMessage } = useIntl();
+  const labelMessage = formatMessage(label);
+
+  return (
+    <Tooltip description={labelMessage}>
+      <Toolbar.ToggleItem value={name} data-state={isActive ? 'on' : 'off'} asChild>
+        <Flex
+          background={isActive ? 'primary100' : ''}
+          padding={2}
+          as="button"
+          hasRadius
+          onMouseDown={(e) => {
+            e.preventDefault();
+            toggleModifier(name);
+          }}
+          aria-label={labelMessage}
+        >
+          <Icon width={4} as={icon} color={isActive ? 'primary600' : 'neutral600'} />
+        </Flex>
+      </Toolbar.ToggleItem>
+    </Tooltip>
+  );
+};
+
+ModifierButton.propTypes = {
+  icon: PropTypes.elementType.isRequired,
+  name: PropTypes.string.isRequired,
+  label: PropTypes.shape({
+    id: PropTypes.string.isRequired,
+    defaultMessage: PropTypes.string.isRequired,
+  }).isRequired,
+};
+
+// TODO: extract a store of modifiers that rules both the toolbar and the leaf renderers
+const modifiers = [
+  {
+    name: 'bold',
+    icon: Bold,
+    label: { id: 'components.Blocks.modifiers.bold', defaultMessage: 'Bold' },
+  },
+  {
+    name: 'italic',
+    icon: Italic,
+    label: { id: 'components.Blocks.modifiers.italic', defaultMessage: 'Italic' },
+  },
+  {
+    name: 'underline',
+    icon: Underline,
+    label: { id: 'components.Blocks.modifiers.underline', defaultMessage: 'Underline' },
+  },
+  {
+    name: 'strikethrough',
+    icon: StrikeThrough,
+    label: { id: 'components.Blocks.modifiers.strikethrough', defaultMessage: 'Strikethrough' },
+  },
+];
```

Comment:
I can't remember was there a reason we left out "code"?

interface StrapiTextNode {
  type: "text";
  text: string;
  bold?: boolean;
  italic?: boolean;
  underline?: boolean;
  strikethrough?: boolean;
  code?: boolean;
}